### PR TITLE
Remove boat interaction event (Fixes #5539)

### PR DESCRIPTION
**Pull request by Owen1212055**
This removes the unneeded interaction event that occurs in ItemBoat. When placing down a boat it would trigger 2 interaction events (1 for placing the boat and 1 for using the item).

This causes placing boats to fire 1 interaction event (from two).

### CODE REVIEW

**Review by electronicboy — APPROVED** [review]
Generally EER on this stuff as it's a behavioural change, but, based on the code, we'd be triggering the exact same form of interaction twice in the event of interacting with a block

**Review by kennytv — APPROVED** [review]
Plz don't merge branches but rebase

**Comment by kennytv** [comment]
Someone remind me/a dev team member to merge this when GH actions come back to life, I don't want to have this in the auto-merge queue for 10 hours and potentially some other commit break this here inbetween ⚰

**Pull request merged**
